### PR TITLE
[Ready] Added aten::symeig operation

### PR DESCRIPTION
**Pull request by vishwakftw**

### CODE REVIEW

**Comment by vishwakftw** [comment]
This should be good to go. I seemed to have clicked the draft option incorrectly. Sorry about that.

File: test/cpp/test_aten_xla_tensor.cpp [hunk 1]
```diff
@@ -666,6 +666,23 @@ TEST_F(AtenXlaTensorTest, TestQR) {
   }
 }
 
+TEST_F(AtenXlaTensorTest, TestSymEig) {
+  static const int dims[] = {4, 7};
+  for (auto m : dims) {
+    at::Tensor a = at::rand({m, m}, at::TensorOptions(at::kFloat));
+    at::Tensor sym_a = a.mm(a.t());
+    auto b = at::symeig(sym_a, /*eigenvectors=*/true, /*upper=*/true);
```
Comment (dlibenzi) [inline test/cpp/test_aten_xla_tensor.cpp]: I'd test for other combinations as well, nesting loops like:
```
for (upper : {true, false})
  for (eigenvectors : {true, false})
```

Comment (vishwakftw) [reply test/cpp/test_aten_xla_tensor.cpp]: Sure, I'll add that. Thank you.

**Comment by vishwakftw** [comment]
@dlibenzi TestSVD is failing for certain configurations of `some` and `compute_uv`, after I added nested loops for that as well. Should I retain them?

**Comment by dlibenzi** [comment]
> @dlibenzi TestSVD is failing for certain configurations of `some` and `compute_uv`, after I added nested loops for that as well. Should I retain them?

Leave SVD alone.
There is no "unique" SVD decomposition (differences in sign in the column vectors), so that is OK.

**Comment by vishwakftw** [comment]
I thought so too. I believe the failures are due to sampling rank deficient matrices.

**Comment by vishwakftw** [comment]
@dlibenzi this should be good to go if you think the code’s OK.

**Pull request merged**